tests/python-gpu/test_gpu_prediction.py

```diff
@@ -185,3 +193,24 @@ def predict_df(x):
 
         for i in range(10):
             run_threaded_predict(X, rows, predict_df)
+
+    @given(strategies.integers(1, 200),
+           tm.dataset_strategy, shap_parameter_strategy, strategies.booleans())
+    @settings(deadline=None)
+    def test_shap(self, num_rounds, dataset, param, all_rows):
+        param.update({"predictor": "gpu_predictor", "gpu_id": 0})
+        param = dataset.set_params(param)
+        dmat = dataset.get_dmat()
+        bst = xgb.train(param, dmat, num_rounds)
+        if all_rows:
+            test_dmat = xgb.DMatrix(dataset.X, dataset.y, dataset.w, dataset.margin)
+        else:
+            test_dmat = xgb.DMatrix(dataset.X[0:1, :])
```

Comment:
Wouldn't it be better to take a subsample with multiple rows, instead of picking the first row?

Reply:
This is for hypothesis to minimise failing examples to a single row. For example if I break something major while developing a can examine the result for a very simple example.

Reply:
Got it.